cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst

```diff
@@ -0,0 +1,134 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :description: Cask Data Application Platform Wikipedia Pipeline Application
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _examples-wikipedia-data-pipeline:
+
+==================
+Wikipedia Pipeline
+==================
+
+A Cask Data Application Platform (CDAP) example demonstrating a typical data processing pipeline using CDAP Workflows.
+
+Overview
+========
+
+This example demonstrates a CDAP application performing analysis on Wikipedia data using MapReduce and Spark programs
+running within a CDAP Workflow - *WikipediaPipelineWorkflow*.
+
+This example can be run in both online and offline modes. In the online mode, the custom action *DownloadWikiDataAction* reads the *pageTitleStream*
+in which each event is one element from the output of the facebook likes API - https://developers.facebook.com/docs/graph-api/reference/v2.4/object/likes
+For each event, it tries to download Wikipedia data for the page using the https://www.mediawiki.org/wiki/API:Main_page API.
+It stores the downloaded data in the KeyValueTable dataset *wikiData*.
+In the offline mode, it expects Wikipedia data formatted per the output of the MediaWiki API above in the stream *wikiStream*.
+The MapReduce program *wikiDataToDataset* consumes this stream and stores it in the same KeyValueTable dataset *wikiData*.
+Data can be uploaded to the *wikiStream* using the CDAP CLI.
+
+Once raw Wikipedia data is available using in either online or offline modes, the *WikipediaPipelineWorkflow* runs a MapReduce program
+*WikiContentValidatorAndNormalizer* that filters bad records from the raw data, as well as normalizes it by converting the wikitext-formatted
+data to plain text. It then stores the output in another KeyValueTable dataset *normalized*
+
+The *WikipediaPipelineWorkflow* then contains a fork, with two branches in parallel. One branch runs the Apache Spark program *SparkWikipediaAnalyzer*.
+This program consumes normalized data and runs topic modeling on it using the Latent Dirichlet Allocation (LDA) algorithm. It stores its
+output in the CDAP Table dataset *lda*, with one row for each iteration, and a column per topic containing the score for that topic.
+The other branch contains a MapReduce program *TopNMapReduce* that consumes the normalized data and produces the top N words in the dataset *topn*
+
+Let's look at some of these components, and then run the application and see the results.
+
+The WikipediaPipeline Application
+---------------------------------
+
+As in the other `examples <index.html>`__, the components
+of the application are tied together by the class ``WikipediaPipelineApp``:
+
+.. literalinclude:: /../../../cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineApp.java

```

Comment:
As this is an included file, add a hash of this file to the build.sh for the examples:
```
  test_an_include 333dc3ee2fd7b5ae36b640a5e34ba3c9 ../../cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineApp.java
  test_an_include 23d3a5c9f8cbe1a41fe706c6f95bad73 ../../cdap-examples/WordCount/src/main/java/co/cask/cdap/examples/wordcount/WordCount.java
}
```
I tested this with the branch, and this is the hash of the file.